packages/local-environment/src/domain.ts

```diff
@@ -41,27 +49,45 @@ export class NomadDomain {
       watcher: signer,
       recoveryManager: signer,
     });
+
+  }
+
+  static newHardhatNetwork(name: string, domain: number, creationParams?: DomainOptions): NomadDomain {
+    const network = new HardhatNetwork(name, domain, creationParams?.forkurl, creationParams?.weth, creationParams?.docker);
+    return new NomadDomain(network, creationParams?.nomadEnv);
   }
 
-  addNomadEnv(e: NomadEnv) {
+  addNomadEnv(e: NomadEnv): void {
     this.nomadEnv = e;
   }
 
   get name(): string {
     return this.network.name;
   }
 
-  ensureAgents(metricsPort = 9090) {
-    if (!this.agents) this.agents = new Agents(this, metricsPort, this.nomadEnv);
+  ensureAgents(metricsPort = 9090, docker?: Dockerode): void {
+    if (!this.agents) this.agents = new Agents(this, metricsPort, docker, this.nomadEnv);
   }
 
-  connectNetwork(d: NomadDomain) {
-    if (!this.connections().includes(d.network.name))
+  connectDomain(d: NomadDomain): void {
+    if (!this.connections().includes(d.network.name) && d.name != this.network.name) {
       this.connectedNetworks.push(d);
+      if (d.connectInReturn(this)) {
+         d.connectDomain(this);
+      }
+    }
   }
 
-  async isAgentsUp(): Promise<boolean> {
-    return await this.agents!.isAllUp();
+  connectInReturn(d: NomadDomain): boolean {
+    if(this.connections().includes(d.network.name)) {
+      return false;
+    }
+    else return true;
+  }
```

Comment:
no, need for this. Please, could u try to do it as it was in my example with additional param (that has default value)

Reply:
Revised